### PR TITLE
[thci] output commissioning log and add thci interface to get it

### PR DESCRIPTION
**Pull request by wgtdkp**
This Adds a new API `getCommissioningLogs` to return decrypted contents of `JOIN_FIN.req/rsp` to the THCI program.

See the same THCI API in openthread: https://github.com/openthread/openthread/blob/002cc8e912a50abda089998c55c8c7f731b5cbf6/tools/harness-thci/OpenThread_WpanCtl.py#L2123

### CODE REVIEW

**Review by Moandor-y — APPROVED** [review]
LGTM

File: src/library/joiner_session.cpp [hunk 1]
```diff
@@ -207,8 +208,17 @@ Error JoinerSession::SendJoinFinResponse(const coap::Request &aJoinFinReq, bool
     joinFin.SetSubType(MessageSubType::kJoinFinResponse);
     SuccessOrExit(error = mCoap.SendResponse(aJoinFinReq, joinFin));
 
+    LOG_INFO(LOG_REGION_JOINER_SESSION, "session(={}) sent JOIN_FIN.rsp: accepted={}", static_cast<void *>(this),
```
Comment (simonlingoogle) [inline src/library/joiner_session.cpp:211]: Could it be a security pitfall to print these messages as logs?

File: tools/commissioner_thci/commissioner_impl.py [hunk 2]
```diff
@@ -401,6 +404,34 @@ def getCOM_TOK(self):
         result = self._execute_and_check('token print')
         return self._hex_to_bytes(result[0])
 
+    def getCommissioningLogs(self):
+        processed_logs = []
+        thci_logs = self._command("grep \"\\[ thci \\]\" {}".format(
+            self.log_file))
```
Comment (simonlingoogle) [inline tools/commissioner_thci/commissioner_impl.py:410]: how do you clean `log_file` if we are running tests one by one?

File: tools/commissioner_thci/commissioner_impl.py [hunk 3]
```diff
@@ -401,6 +404,34 @@ def getCOM_TOK(self):
         result = self._execute_and_check('token print')
         return self._hex_to_bytes(result[0])
 
+    def getCommissioningLogs(self):
```
Comment (simonlingoogle) [inline tools/commissioner_thci/commissioner_impl.py:407]: `getCommissioningLogs` needs to wait for the commissioning process to finish. How do you make sure of that?

File: src/library/joiner_session.cpp (hunk 1 above)
Comment (wgtdkp) [reply src/library/joiner_session.cpp:211]: What about make this optional with a compile flag?

File: tools/commissioner_thci/commissioner_impl.py (hunk 2 above)
Comment (wgtdkp) [reply tools/commissioner_thci/commissioner_impl.py:410]: The log file is overridden for each run of the commissioner.

File: tools/commissioner_thci/commissioner_impl.py (hunk 3 above)
Comment (wgtdkp) [reply tools/commissioner_thci/commissioner_impl.py:407]: I think we have no way to control when the method is called. The caller must know when to call this method.

File: src/library/joiner_session.cpp (hunk 1 above)
Comment (simonlingoogle) [reply src/library/joiner_session.cpp:211]: Agree that a compile flag is needed. Just like in OT, we only print these logs when REFERENCE_DEVICE is ON.

File: tools/commissioner_thci/commissioner_impl.py (hunk 3 above)
Comment (simonlingoogle) [reply tools/commissioner_thci/commissioner_impl.py:407]: Not sure if that's true of TestHarness. We don't know how `getCommissioningLogs` is used by TestHarness.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -47,6 +47,7 @@ jobs:
                 -DCMAKE_BUILD_TYPE=Release        \
                 -DCMAKE_INSTALL_PREFIX=/usr/local \
                 -DOT_COMM_COVERAGE=ON             \
+                -DOT_COMM_COMMISSIONING_LOG=ON    \
```
Comment (simonlingoogle) [inline .github/workflows/tests.yml:50]:
```suggestion
                -DOT_COMM_REFERENCE_DEVICE=ON    \
```

OT used to use `CERT_LOG` for certification logs, but later changed to `REFERENCE_DEVICE`.
Should we be consistent with OT and OTBR here?

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:
with a nit

File: .github/workflows/tests.yml (hunk 4 above)
Comment (wgtdkp) [reply .github/workflows/tests.yml:50]: Agree, will make the change accordingly.

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -47,6 +47,7 @@ jobs:
                 -DCMAKE_BUILD_TYPE=Release        \
                 -DCMAKE_INSTALL_PREFIX=/usr/local \
                 -DOT_COMM_COVERAGE=ON             \
+                -DOT_COMM_REFERENCE_DEVICE=ON    \
```
Comment (jwhui) [inline .github/workflows/tests.yml:50]: nit: alignment

File: CMakeLists.txt [hunk 6]
```diff
@@ -29,13 +29,14 @@
...
-option(OT_COMM_APP              "Build the CLI App" ON)
-option(OT_COMM_CCM              "Build with Commercial Commissioning Mode" ON)
-option(OT_COMM_COVERAGE         "Enable coverage reporting" OFF)
-option(OT_COMM_JAVA_BINDING     "Build Java binding" OFF)
-set(OT_COMM_JAVA_BINDING_OUTDIR "" CACHE STRING "Specify output directory of generated Java source files")
-option(OT_COMM_TEST             "Build tests" ON)
+option(OT_COMM_ANDROID           "Build with Android NDK" OFF)
+option(OT_COMM_APP               "Build the CLI App" ON)
+option(OT_COMM_CCM               "Build with Commercial Commissioning Mode" ON)
+option(OT_COMM_REFERENCE_DEVICE "Enable features for being a Thread Group reference device" OFF)
```
Comment (jwhui) [inline CMakeLists.txt:35]: nit: alignment

**Review by jwhui — APPROVED** [review]

**Pull request merged**